### ISSUE TITLE
Dragons can’t breed [1.12.2]

### ISSUE THREAD

**Issue by itsfriday19** [Alex-the-666/Ice_and_Fire#2231]
I’m experiencing an issue where my fire dragons aren’t breeding (haven’t tried with ice dragons). They’re both stage 4. I’m giving the male the fire lily mixture, then the female. They get hearts for a bit but then nothing happens. I made sure they were close to each other and on even ground (if that makes a difference), and tried having them set to wander, stay, or escorting, none of it has worked. 
I’ve tried looking for similar bug reports for this version of the mod (1.8.4) but haven’t been able to find any, so I don’t know if it’s just me.

**Comment by SilPan** [comment]
I have observed the same Problem, with 1.8.3 and 1.8.4.

At some point it I thought it was related to the state the Dragon is in [stay|escort|wandering], but that turned out to be not true.

I have tried to reproducibly describe what causes this, because at some point most Dragons were able to breed. And now they don't any more. I ended up assuming that the dragons get stuck in some kind of internal state where they don't get out of any more. I tried exploring the Dragons NBT tags but nothing looked suspicious. I also went through the source code and took a look at the Dragon Classes trying to understand the internal state machines, if these can get trapped in some state somehow, but that did not give me a lead either.

I have also tested it with an old world from version I don't remember upwards and a new one, behaviour is similarly weird.

I feels very indeterministic. Makes me uncomfortable.
